[Bug]: Issue with `AzureBlobStore`  in maggma 0.73.0

### Code snippet

```python

```

### What happened?

@rkingsbury, I happened to just update maggma the second you did, and I am getting the following error when importing jobflow. Should I open an issue on jobflow or is this a maggma-specific issue? Looks like a type hinting issue in maggma?

```python
______________ ERROR collecting tests/jobflow/test_copy_files.py _______________
tests/jobflow/test_copy_files.py:5: in <module>
    jf = pytest.importorskip("jobflow")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:999: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/__init__.py:11: in <module>
    from jobflow.settings import JobflowSettings
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/settings.py:7: in <module>
    from maggma.stores import MemoryStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/__init__.py:6: in <module>
    from maggma.stores.azure import AzureBlobStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:56: in <module>
    class AzureBlobStore(Store):
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:392: in AzureBlobStore
    def _get_container(self) -> ContainerClient | None:
                                ^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
______________ ERROR collecting tests/jobflow/test_customizers.py ______________
tests/jobflow/test_customizers.py:5: in <module>
    jf = pytest.importorskip("jobflow")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:999: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/__init__.py:11: in <module>
    from jobflow.settings import JobflowSettings
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/settings.py:7: in <module>
    from maggma.stores import MemoryStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/__init__.py:6: in <module>
    from maggma.stores.azure import AzureBlobStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:56: in <module>
    class AzureBlobStore(Store):
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:392: in AzureBlobStore
    def _get_container(self) -> ContainerClient | None:
                                ^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
______________ ERROR collecting tests/jobflow/test_emt_recipes.py ______________
tests/jobflow/test_emt_recipes.py:5: in <module>
    jf = pytest.importorskip("jobflow")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:999: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/__init__.py:11: in <module>
    from jobflow.settings import JobflowSettings
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/settings.py:7: in <module>
    from maggma.stores import MemoryStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/__init__.py:6: in <module>
    from maggma.stores.azure import AzureBlobStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:56: in <module>
    class AzureBlobStore(Store):
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:392: in AzureBlobStore
    def _get_container(self) -> ContainerClient | None:
                                ^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
________________ ERROR collecting tests/jobflow/test_syntax.py _________________
tests/jobflow/test_syntax.py:5: in <module>
    jf = pytest.importorskip("jobflow")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:999: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/__init__.py:11: in <module>
    from jobflow.settings import JobflowSettings
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/settings.py:7: in <module>
    from maggma.stores import MemoryStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/__init__.py:6: in <module>
    from maggma.stores.azure import AzureBlobStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:56: in <module>
    class AzureBlobStore(Store):
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:392: in AzureBlobStore
    def _get_container(self) -> ContainerClient | None:
                                ^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
_______________ ERROR collecting tests/jobflow/test_tutorials.py _______________
tests/jobflow/test_tutorials.py:5: in <module>
    jf = pytest.importorskip("jobflow")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:999: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/__init__.py:11: in <module>
    from jobflow.settings import JobflowSettings
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/jobflow/settings.py:7: in <module>
    from maggma.stores import MemoryStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/__init__.py:6: in <module>
    from maggma.stores.azure import AzureBlobStore
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:56: in <module>
    class AzureBlobStore(Store):
/opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/maggma/stores/azure.py:392: in AzureBlobStore
    def _get_container(self) -> ContainerClient | None:
                                ^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
=========================== short test summary info ============================
ERROR tests/jobflow/test_copy_files.py - TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
ERROR tests/jobflow/test_customizers.py - TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
ERROR tests/jobflow/test_emt_recipes.py - TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
ERROR tests/jobflow/test_syntax.py - TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
ERROR tests/jobflow/test_tutorials.py - TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
!!!!!!!!!!!!!!!!!!! Interrupted: 5 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 5 errors in 2.72s ===============================
```


### Version

0.73.0

### Which OS?

- [ ] MacOS
- [ ] Windows
- [ ] Linux

### Log output

```shell

```

--- a/src/maggma/stores/azure.py
+++ b/src/maggma/stores/azure.py
@@ -2,17 +2,21 @@
 Advanced Stores for connecting to Microsoft Azure data.
 """
 
+from __future__ import annotations
+
 import importlib
 import os
 import threading
 import warnings
 import zlib
-from collections.abc import Iterator
 from concurrent.futures import wait
 from concurrent.futures.thread import ThreadPoolExecutor
 from hashlib import sha1
 from json import dumps
-from typing import Literal
+from typing import TYPE_CHECKING, Literal
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
 
 import msgpack  # type: ignore
 from monty.msgpack import default as monty_default

